src/codegen/doc/format.rs

```diff
@@ -138,7 +140,7 @@ fn replace_symbols(
     env: &Env,
     in_type: Option<(&TypeId, Option<LocationInObject>)>,
 ) -> String {
-    if env.config.use_gi_docgen {
+    if env.library.doc_format == DocFormat::GiDocgen {
```

Comment:
Perhaps for now this could be a `if library.doc_format == DocFormat::GiDocgen || config.use_gi_docgen` for the moment.

Reply:
that is an option yes, although, it will end up staying there given no one will remember to remove it. As this option only impacts docs generation, i rather have bindings manually amend the format into the gir file then gir keeping this old "hack"

Reply:
My fear is that this is a bit of a chicken and egg situation. Breaking gir generation is a discouragement for setting doc_format and this MR depends on libraries using it. 

I suppose an alternative is to have a prequel for this PR that just reads the doc_format and does nothing with it (instead of crashing) to unblock the situation.

Reply:
It is always been like that when it comes to new annotations so i rather fix things properly.